kafka-client/src/main/java/dev/responsive/kafka/internal/utils/SessionUtil.java

```diff
@@ -99,45 +95,18 @@ public static CqlSession connect(
   }
 
   public static MongoClient connect(
-      final String hostname,
-      @Nullable final String clientId,
-      @Nullable final String clientSecret,
+      final String connectionString,
       final String additionalParams,
       @Nullable final ResponsiveMetrics metrics
   ) {
-    final String connectionString;
-    if (clientId != null && clientSecret != null) {
-      // TODO(agavra): consider allowing configuration of the read consideration
-      // some situations (such as non-EOS) could benefit from performance of non-majority
-      // reads if they're ok with the risks
-      connectionString = String.format(
-          "mongodb+srv://%s:%s@%s/?retryWrites=true&w=majority&r=majority",
-          URLEncoder.encode(clientId, StandardCharsets.UTF_8),
-          URLEncoder.encode(clientSecret, StandardCharsets.UTF_8),
-          hostname
-      );
-    } else if (clientId == null ^ clientSecret == null) {
-      throw new IllegalArgumentException(
-          "Must specify both or neither Mongo username and password.");
-    } else {
-      // TODO(agavra): TestContainers uses a different connection string, for now
-      // we just assume that all non authenticated usage is via test containers
-      connectionString = hostname;
-    }
-
     final String connectionStringWithParams = additionalParams.equals("")
         ? connectionString
         : connectionString + "/?" + additionalParams;
 
-    ServerApi serverApi = ServerApi.builder()
-        .version(ServerApiVersion.V1)
-        .build();
-
     final MongoClientSettings.Builder settingsBuilder = MongoClientSettings.builder()
-        .applyConnectionString(new ConnectionString(connectionStringWithParams))
+        .applyConnectionString(new ConnectionString(connectionString))
         .readConcern(ReadConcern.MAJORITY)
-        .writeConcern(WriteConcern.MAJORITY)
-        .serverApi(serverApi);
```

Comment:
this is the line that was removed for DocumentDB compatibility - not sure why it wasn't supported